src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -532,27 +532,39 @@ def inner(*args, **kwargs):
                         or isinstance(value, str):
                     kwargs[key] = unquote(value)
 
+            # Process method arguments.
             if method in ['GET', 'DELETE']:
-                ret = func(*args, **kwargs)
-
+                pass
             elif cherrypy.request.headers.get('Content-Type', '') == \
                     'application/x-www-form-urlencoded':
-                ret = func(*args, **kwargs)
-
+                pass
             else:
                 content_length = int(cherrypy.request.headers['Content-Length'])
                 body = cherrypy.request.body.read(content_length)
-                if not body:
-                    ret = func(*args, **kwargs)
-                else:
+                if body:
                     try:
                         data = json.loads(body.decode('utf-8'))
                     except Exception as e:
                         raise cherrypy.HTTPError(400, 'Failed to decode JSON: {}'
                                                  .format(str(e)))
                     kwargs.update(data.items())
-                    ret = func(*args, **kwargs)
 
+            # Audit API request.
```

Comment:
There is a better place for that: `tools.RequestLoggingTool`.

Reply:
I know, and this was my first approach, but in this case the method arguments are not available by default and i had to extract them the same way as in ``controllers/__init__.py``. This would mean duplicate code or i had to refactor and extract the necessary code part to be able to use it in ``tools.RequestLoggingTool``. Why don't you like the implementation as it is now? The result is the same and i do not see any other reasons against that.

Reply:
Mainly because we have now three independent places to log requests:

1. CherryPy already logs requests to the default logger
2. `tools.RequestLoggingTool` is logging requests, too.
3. Your implementation implements a third way to log requests. 

Having three implementations doing the same thing feels wrong. 

Also it's a matter of separation of logic into different classes.